portfolio/src/main/webapp/script.js

```
@@ -57,12 +57,51 @@ async function setHelloContainerWithServlet() {
  function setMyTeamConatainerWithServerlet(){
   fetch('/Team').then(response => response.json()).then((theTeam) => {
 
-    var teamParagraph = 'Here are the members of my team and their position:\n';
+    var teamParagraph = 'Here are the members of my team and their position:\n\n';
 
     for (i = 0; i < theTeam.members.length; i++){
         teamParagraph =  teamParagraph + theTeam.members[i].name + '\t'+ theTeam.members[i].status + '\n'
     }
 
     document.getElementById('team-container').innerText = teamParagraph;
  });
-}
\ No newline at end of file
+}
+
+function setCommentContainerWithServlet() {
+  fetch('/comments').then(response => response.json()).then((commentPackage) => {
+
+    console.log("Does this run???");
+
+    var commentParagraph = 'Here are the comments:\n\n';
+
+    const commentContainer = document.getElementById('comment-container');
+    var validateComments = commentPackage.flatMap(comment => validateComment(comment, x => [x], x => {console.log('Invalid comment ', x); return [];}) );
```

It's great to define anonymous, in-line functions like this where it aids readability. When you do so, please use multi-line block syntax where doing so makes sense.

For example, this would be easier to read as:

```
var validateComments = commentPackage.flatMap(comment => validateComment(
    comment,
    x => [x],
    x => {
        console.log('Invalid comment ', x);
        return [];
    }));
```

However, it would be even clearer than that to use an explicit `filter` rather than a `flatMap` (since you're not actually transforming the values here, but just passing them through directly via `x => [x]`:

```
let validComments = commentPackage.filter(comment => {
    if (comment.name === "" || comment.payload === "") {
        console.log("Invalid comment ", comment);
        return false;
    }
    return true;
});
validComments.map(formatComment)...;
```

In fact, if you do validation in the server and trust the output, you theoretically shouldn't even need to filter comments this way. Doing so is a trade-off, but it might make sense to add validation to the server now since you'll be storing data in Datastore later and you most likely don't want to persist invalid comments.

---

This is code I helped him write. I still think client side validation is important so errors are explicitly handled instead of silent failures when something is wrong. I agree that switching validateComment to a boolean and filtering/unzipping might be nice, but I was just trying to show some javascripty ways to do things, which sometimes involves using callbacks like this.